### PR TITLE
clone spec for each particle-recipe

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

**Comment by mmandlis** [comment]
please, hold on reviewing this, i'm still making changes. sorry.

**Comment by mmandlis** [comment]
Ready for review, please, take a look. Thanks!

File: runtime/particle-spec.js [hunk 1]
```diff
@@ -112,17 +113,29 @@ class ParticleSpec {
...
+  static fromLiteral(literal, typeVarMap) {
     let {args, name, verbs, transient, description, implFile, affordance, slots} = literal;
-    args = args.map(({type, direction, name}) => ({type: Type.fromLiteral(type), direction, name}));
-    return new ParticleSpec({args, name, verbs, transient, description, implFile, affordance, slots});
+    args = args.map(({type, direction, name, isOptional}) => ({type: Type.fromLiteral(type), direction, name, isOptional}));
+    return new ParticleSpec({args, name, verbs, transient, description, implFile, affordance, slots}, typeVarMap);
+  }
+
+  clone() {
+    let typeVarMapClone = new Map();
```
Comment (dstockwell) [inline runtime/particle-spec.js]: I'm a bit stuck on this bit, are you sure we need to clone the type var map? It seems to me that constructing the particle spec, constructs connection specs which will then repopulate the map.

Comment (mmandlis) [reply runtime/particle-spec.js]: I got rid of it for now. 
Instead, i added cloning of the hosted particle recipe in manifest parsing.
I'll merge it for now, and try to make progress with the other PR (the type resolution) and see if the errors go away.

**Pull request merged**